modules/tinymce/src/plugins/table/main/ts/ui/TableDialog.ts

```diff
@@ -107,10 +107,6 @@ const onSubmitTableForm = (editor: Editor, tableElm: HTMLTableElement | null | u
 
   api.close();
 
-  if (data.class === '') {
-    delete data.class;
-  }
```

Comment:
I am not sure what should happen if the table has a class already on it that is not in the dialog class list. By removing the above code, `<table class="random" ...>` would be removed despite not explicitly changing the class dropdown in the dialog at all to 'None'.

I am pretty sure we have run into this situation before with the cell dialog. We'll have to see what we did there

Reply:
Maybe we need to look at having a base `Select...` option that the dropdown defaults to if the class attribute on the table does not match any of the predefined options.
e.g. const noMatchClass: Dialog.ListBoxItemSpec[] = [{ text: 'Select...', value: 'select' }];

Reply:
@ltrouton table cell dialog deletes the class in that scenario as Dan has done here:
https://fiddle.tiny.cloud/Ky7rRJzEPV/0

Maybe log a new ticket to suggest improving both 🤔

Reply:
@TheSpyder I believe what Luke is saying is that if we add a class not in the `table_class_list` like `class="random"`. Then, we open the Table dialog, make no changes, and then save:
* On 6-stable, the content is not modified as expected;
* On this branch, the `class="random"` is removed.

This is because any class not in the class list would default to the first None value, which would remove the class even though no changes have been made. The suggestion is to make another placeholder value like `Select...` instead of the first value as the fallback when the Table class is not recognised to be in the class list.

Reply:
I think there might be a misunderstanding of what I was referring to in my comment as I think the new behaviour is causing a potential regression from stable. With the change, it is possible for the class attribute to be removed from a table by opening the table dialog, making no changes then clicking save which doesn't seem right.

I'll demo it to Daniel next week and hopefully that will clear up any confusion

Reply:
I do understand that - but what I'm trying to say is this is how the row and cell dialogs already work. Try my fiddle  to confirm.

All we're doing here is applying that behaviour to the table dialog. This is likely expected behaviour based on the bug report where it isn't applying correctly.

If it's undesirable behaviour that seems like something we should discuss for v7?

Reply:
> This scenario is likely what those lines of code were guarding against

I suspect what those lines were trying to do was _implement_ this behaviour.

In the row and cell dialogs, there's a system that compares the input to the dialog value when the dialog is saved - so deleting the value results in it being deleted it from the DOM:
https://github.com/tinymce/tinymce/blob/9c486f7c951e55ae34bb548efe897eec705ac04d/modules/tinymce/src/plugins/table/main/ts/ui/CellDialog.ts#L45-L47

Unfortunately the table dialog was implemented differently so instead these lines of code made the empty string case have no effect.

Reply:
Implemented changes outlined by Luke in the Jira ticket